### PR TITLE
Add a section on when to choose Zend\Expressive over Zend\Mvc

### PR DESCRIPTION
**Pull request by settermjd**
This PR: 

- Adds a section to the documentation on when to choose Zend\Expressive
over Zend\Mvc

I'm creating this PR after participating in the discussion on issue #283, _"When to choose Zend\Expressive over Zend\Mvc"_. After providing my feedback, @weierophinney asked me to take what I'd written and add it to the Zend\Expressive and Zend\Mvc docs. This commit adds said revised version of my feedback, to the docs.

### CODE REVIEW

**Comment by settermjd** [comment]
As I said, this is largely what I contributed to the discussion, just cleaned up, with first person narrative removed. @weierophinney, if you're looking for a much more fleshed out document, which has considerations on the merits of both, along with use cases, I'm happy to expand the document to cater to that. Otherwise, if this suits what you're after, that's good with me.

**Comment by settermjd** [comment]
Actually, on reviewing the page again, I believe it would benefit from a section on when to choose Zend\Mvc. However, I'd need assistance on that, so that Zend\Mvc is given a fair coverage.

File: doc/book/why-expressive.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Should You Choose Zend Expressive Over Zend\Mvc?
+
+After creating several projects with Zend\Expressive, we recommend that you choose Expressive for any new project — _if the choice is yours to make_.
+
+## We Recommend Zend Expressive
```
Comment (froschdesign) [inline doc/book/why-expressive.md]: "Zend\Expressive" vs. "Zend Expressive"

File: doc/book/why-expressive.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+What’s more, it also has several pre-wired structures in place.
+
+As a result, you are required to know a lot of what those things are — *if* you want to use it optimally.
+And to acquire that depth of knowledge, you’re going to need to spend a lot of time, digging deep into Zend\Mvc’s internals, before you begin to get the most out of it.
+
+To quote Zend Framework project lead, [Matthew Weier O’Phinney](https://twitter.com/mwop):
+
+> The problem is that Zend\Mvc is anything but beginner-friendly at this point. You're required to deep dive into the event manager, service manager, and module system — right from the outset; And to do this you need more than a passing understanding of object-oriented programming and a range of design patterns.
+
+Zend Expressive (specifically applications based on [the Expressive Skeleton Installer](https://zendframework.github.io/zend-expressive/)) on the other hand, comes with barely any of these assumptions and requirements.
```
Comment (froschdesign) [inline doc/book/why-expressive.md]: Same here.

File: doc/book/why-expressive.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+- PSR-7 Middleware
+
+Given that, you can quickly get up to speed with the framework and begin creating the application that you need.
+We believe that this approach — in contrast to the Zend\Mvc approach — is more flexible and accommodating.
+
+What’s more, you can mix and match the types of applications that you create.
+Do you just need an API? Great; you can do that quite quickly.
+Do you want an HTML-based front-end? That’s available too.
+
+When building applications with Zend Expressive, you can make use of any other Zend library, or non-Zend library.
```
Comment (froschdesign) [inline doc/book/why-expressive.md]: Same here.

**Comment by settermjd** [comment]
@froschdesign thanks for the feedback. Document updated.

**Comment by froschdesign** [comment]
@weierophinney 
Question: (`zend-expressive` and `zend-mvc`) **or** (`Zend\Expressive` and `Zend\Mvc`) ?

**Comment by settermjd** [comment]
@froschdesign the latter seems to make more sense. @weierophinney?

**Comment by froschdesign** [comment]
@settermjd 
`zend-*` is used in newer documentations. Here are some examples:

* https://docs.zendframework.com/zend-crypt/intro/
* https://docs.zendframework.com/zend-component-installer/
* https://docs.zendframework.com/zend-expressive/getting-started/features/
* https://docs.zendframework.com/zend-hydrator/quick-start/
* https://zendframework.github.io/zend-mvc/cookbook/middleware-in-listeners/

**Comment by weierophinney** [comment]
Popping in to confirm what @froschdesign has noted: we use `zend-{component}` within documentation, and not the namespace. That allows searching on the package name and getting back links to the specific component documentation pages.

**Pull request merged**